modules/tinymce/src/core/main/ts/api/dom/Selection.ts

```diff
@@ -442,9 +441,10 @@ const EditorSelection = (dom: DOMUtils, win: Window, serializer: DomSerializer,
    * // Alerts the currently selected elements node name
    * alert(tinymce.activeEditor.selection.getNode().nodeName);
    */
-  const getNode = (): Element => ElementSelection.getNode(editor.getBody(), getRng());
+  // TODO: Investigate if this type is correct
```

Comment:
Ahh yeah, I added this as something to come back to when I got to strict typing this part of the code. The problem is that it appears as though `ElementSelection.getNode` should be `Node | null`. In my quick manual testing I'm not sure that was actually possible though.

Reply:
I wonder if we should change `ElementSelection.getNode()` to something like this to be 100% sure it's always going to be an element?
```ts
  const elm = NodeType.isText(node) ? node.parentNode : node;
  return NodeType.isElement(elm) ? elm : root;
```

Reply:
yes, looking at this now I think the change of `getNode` to return `Node | null` instead of `Element` is incorrect. Lots of things become easier if it just returns an element.

Reply:
Changed this to use the above as discussed and have removed the TODO/`Element` cast.